Automattic/jetpack
# Podcast Player Block: Add list's front-end styles & UI handling

Related #14942
Built on #14952

#### Changes proposed in this Pull Request:

- [x] Create markup with all necessary fields for the episodes list.
- [x] Add styling according to the design.
- [x] Handle episode play, pause & error states for multiple player instances.
- [x] Add necessary aria labels. (cc @jeryj)

#### Podcast Player block's episode list design:

- Episode playback:
   <img width="297" alt="Screenshot 2020-03-16 19 44 31" src="https://user-images.githubusercontent.com/1451471/76790181-a4875080-67be-11ea-9445-5fecad45801b.png">
- Episode playback error:
   <img width="297" alt="Screenshot 2020-03-16 19 44 19" src="https://user-images.githubusercontent.com/1451471/76790179-a2bd8d00-67be-11ea-8529-8955b0837216.png">
- Icons:
   - https://material.io/resources/icons/?icon=volume_up&style=outline
   - https://material.io/resources/icons/?icon=error_outline&style=baseline
- Font family:
   `-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;`

#### Testing instructions:

1. Add couple Podcast Player blocks to your page
   Example links:
   - https://anchor.fm/s/9400d7c/podcast/rss
   - https://anchor.fm/s/807f8e8/podcast/rss
2. Open the page preview / publish.
3. Click on any episode to start playback. A playing episode should have a loudspeaker icon on its left.
4. If an episode from another Player is clicked, the current one should be paused. A paused episode should not have any icon on its left but should keep the font bold.
5. If an active episode is clicked, it should pause/resume playing.
7. When episode error occurs, a message should appear below that episode (as per design). The icon on the left should be red, but due to [some issues with SVG](https://github.com/Automattic/jetpack/pull/14989#discussion_r394262451) it is black and will be dealt with in a follow-up PR.
   _Tip: Episode error can be tested by [blocking the request](https://developers.google.com/web/updates/2017/04/devtools-release-notes#block-requests) to its URL. You can find the URL to block assigned to the `data-jetpack-podcast-audio` attribute._ 

#### Notes for Reviewers:

1. Player's controls events are not yet handled, so e.g. pausing the episode via the play/pause button won't have any visual impact on the episode list items. This will be addressed in a follow-up PR._
2. The JS implementation is being majorly overhauled into its own class here: https://github.com/Automattic/jetpack/pull/15034 Getting this PR merged would make work on improving this implementation a lot easier. We're not looking for anything close to perfection on this PR for this beta block.

#### Proposed changelog entry for your changes:

- Podcast Player Block: Add episodes list's front-end styles & UI handling
